payment-in-stablecoins/README.md

```diff
@@ -0,0 +1,95 @@
+# Proposal Payment in Stablecoins
+
+## Simple Summary
+
+Allows the DAO to execute proposals that send funds denominated in USD. This change addresses the ETH/USD price volatility risk both sides suffer from today, as well as the DAO's ability to perform ETH-to-USD swaps trustlessly.
+
+## Abstract
+
+This specification introduces two new contracts that facilitate stablecoin payments:
+
+- `NOU` a new ERC20 token the DAO can mint to say "I owe you this amount of USD"
+- `NounsStablecoinPayments` a contract that:
+  - allows the DAO to mint NOU tokens as payment to proposal recipients
+  - allows anyone to sell it stablecoins in exchange for ETH with some additional incentive
+  - allows proposal recipients to redeem NOU for stablecoins
+
+## Technical Specification
+
+### Overview
+
+#### `NOU` contract
+
+A new ERC20 token contract:
+
+- Inherits from OpenZeppelin's `AccessControlEnumerable`
+  - Assigns the `ADMIN_ROLE` to an `admin` provided in the constructor; admin can administer its own role and all other roles
+- `function mint(address recipient, uint amount)`
+  - Mints `amount` tokens to `recipient`
+  - Permits only senders of role `MINTER_ROLE`
+- `function burn(address from, uint amount)`
+  - Burns `amount` tokens from `from`'s account
+  - Permits only senders of role `BURNER_ROLE`
+
+#### `NounsStablecoinPayments` contract
+
+A new contract that helps the DAO acquire stablecoins by buying them from bots for a margin, and helps proposal recipients redeem stablecoin payments.
+
+##### Configuration variables
+
+- `Stablecoin` the address of the ERC20 token to use as USD, e.g. USDC or DAI
+- `BotIncentiveBPs` the basis points to add on top of the oracle price to further incentivize bots to sell USD to this contract
+- `Oracle` the Chainlink Stablecoin/ETH oracle address
+- `NOU` the address of the NOU token ERC20 contract
+- `Admin` the address of the admin that can withdraw this contract's balances
+- `FundingRate` the value multiplier that determines how much ETH funding is required to mint NOU tokens; e.g. if set to 2.5, and the DAO wants to mint 100 NOU, it would need to send ETH worth 250 USD
+
+##### View functions
+
+- `function stablecoinAmountNeeded()` returns the amount of USD the DAO needs to buy, should be the same as `NOU.totalSupply() - Stablecoin.balanceOf(this)`
+- `function price(uint usdAmount)` returns the amount of ETH this contract is willing to pay for the given `usdAmount`, taking into account oracle pricing and the additional bot incentive factor
+- `function stablecoinBalance()` returns the contract's balance in the desired stablecoin
+
+##### `sellStablecoin` transaction
+
+- `function sellStablecoin(uint amount)` sell this contract stablecoins in exchange for ETH at the exchange rate determined by the `price` function
+
+Rules
```

Comment:
another rule:
* this will fail if the contract is not holding enough ETH to send to msg.sender